server/src/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1078,6 +1082,16 @@ public void cleanupStorage(boolean recurring) {
                         }
                     }
 
+                    //destroy snapshots in destroying state in snapshot_store_ref
+                    List<SnapshotDataStoreVO>  ssSnapshots = _snapshotStoreDao.listByState(ObjectInDataStoreStateMachine.State.Destroying);
+                    for(SnapshotDataStoreVO ssSnapshotVO : ssSnapshots){
+                        try {
+                            _snapshotService.deleteSnapshot(snapshotFactory.getSnapshot(ssSnapshotVO.getSnapshotId(), DataStoreRole.Image));
```

Comment:
@koushik-das If there is any failure then the db entry will remain Destroying and will only be updated if there is successful deletion. Condition which is checking Destroying state of object  in SnapshotObject class is meant for this purpose.